scylla/src/client/session_builder.rs

```diff
@@ -590,10 +554,8 @@ impl<K: SessionBuilderKind> GenericSessionBuilder<K> {
     /// # Ok(())
     /// # }
     /// ```
-    pub async fn build(
-        &self,
-    ) -> Result<GenericSession<CurrentDeserializationApi>, NewSessionError> {
-        GenericSession::connect(self.config.clone()).await
+    pub async fn build(&self) -> Result<Session, NewSessionError> {
+        Session::connect(self.config.clone()).await
     }
```

Comment:
:pushpin: Why is `build` borrowing instead of consuming the `SessionBuilder`? This makes no sense, as `SessionBuilder` implements `Clone`, so it could always be cloned before building to retain it. Now there is needless cloning for the usual case.

Reply:
I can change that in the follow-up in order to not expand the scope of this PR